core/types.ml

```diff
@@ -1692,10 +1909,10 @@ struct
       let bound_vars, vars =
         List.fold_left
           (fun (bound_vars, vars) tyvar ->
-             let var, spec = varspec_of_tyvar tyvar in
-               (TypeVarSet.add var bound_vars, (var, spec)::vars)) (bound_vars, []) tyvars in
-        (bound_vars, List.rev vars) in
-
+            let var, spec = varspec_of_tyvar tyvar in
+             (TypeVarSet.add var bound_vars, (var, spec)::vars)) (bound_vars, []) tyvars in
```

Comment:
indentation seems off here; putting the `in` at the end on the next line (or unnesting the `let`s a bit) might also help readability